pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -238,11 +235,28 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr model.Config, event mo
 	}
 
 	if fullPush {
+		// When doing a full push, for added and updated services trigger an eds update
+		// so that endpoint shards are updated.
+		var instances []*model.ServiceInstance
+		if len(addedSvcs) > 0 {
+			instances = append(instances, convertInstances(curr, addedSvcs)...)
+		}
+		if len(updatedSvcs) > 0 {
+			instances = append(instances, convertInstances(curr, updatedSvcs)...)
+		}
+		s.edsUpdate(instances)
```

Comment:
Does it need an update for unchanged svc here?

Reply:
https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/serviceentry/servicediscovery.go#L198 - Unchanged svs are handled here. This is for added/updated svcs only

Reply:
Unchanged svc doesn't mean its endpoints are unchanged, right?

Reply:
I see. Yes. Good point

Reply:
Looking at again, for unchangedSvcs, it comes to full push only for DNS type of resolution https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/serviceentry/servicediscovery.go#L228 - So we do not need endpoint shard update for them

Reply:
I can not understand all. We donot update eds during full push for k8s registry, right?

Reply:
Yes. Correct. But endpoint update events in K8s will keep the EndpointShards in sync with registry. But in case of service entry, when we do full push we do not update EndpointShards making it stale and only `UpdateServiceShards` call during full push refreshes the data.

Reply:
Added unchangedsvcs as per offline discussion